app/Http/Controllers/Api/ExperienceSkillsController.php

```diff
@@ -12,13 +12,26 @@ class ExperienceSkillsController extends Controller
     public function store(StoreExperienceSkill $request)
     {
         $data = $request->validated();
-        $experienceSkill = new ExperienceSkill($data);
-        $experienceSkill->skill_id = $data['skill_id'];
-        $experienceSkill->experience_id = $data['experience_id'];
-        $experienceSkill->experience_type = $data['experience_type'];
-        $experienceSkill->justification = $data['justification'];
-        $experienceSkill->save();
-        return new JsonResource($experienceSkill->fresh());
+        // Restore soft deleted experienceSkill if it exists, otherwise create a new one.
+        $softDeletedExperienceSkill = ExperienceSkill::onlyTrashed()
+            ->where([
+            ['skill_id', $data['skill_id']],
+            ['experience_id', $data['experience_id']],
+            ['experience_type', $data['experience_type']]
+        ])->first();
```

Comment:
Assuming the ExperienceSkill ID is returned with the `$request` it's probably safer/faster to use that to check for soft deleted models

Reply:
I agree, however it's not. The id is auto incremented when a new experience skill is created.

Reply:
Ah gotcha. Sorry I thought `store` was being used for creates and updates.